### PR TITLE
[DF] Fix dataframe_concurrency test on 32bit machines

### PR DESCRIPTION
**Pull request by stwunsch**
If we use more than 2 GB of memory on a 32bit Windows machine, the test
will break because of the memory limit. This happens if we run the tests
with a high concurrency, e.g., with 32 threads. Fix the concurrency to 8
threads to prevent this issue.

### CODE REVIEW

File: tree/dataframe/test/dataframe_concurrency.cxx [hunk 1]
```diff
@@ -10,10 +10,12 @@
 
 #include "gtest/gtest.h"
 
+static const auto NUM_THREADS = 8u;
+
 #ifdef R__USE_IMT
 TEST(RDFConcurrency, NestedParallelismBetweenDefineCalls)
 {
-   ROOT::EnableImplicitMT();
+   ROOT::EnableImplicitMT(NUM_THREADS);
```
Comment (pcanal) [inline tree/dataframe/test/dataframe_concurrency.cxx:18]: It is not useless to run with the max number of core when we can.  Could we make this to be set just for the 32 bits build?

Comment (stwunsch) [reply tree/dataframe/test/dataframe_concurrency.cxx:18]: Yep, I can do that!

Comment (stwunsch) [reply tree/dataframe/test/dataframe_concurrency.cxx:22]: @pcanal What about this?

Comment (pcanal) [reply tree/dataframe/test/dataframe_concurrency.cxx:22]: Wouldn't the test also fails on Linux/32bits ... if there was enough cores?

**Comment by Axel-Naumann** [comment]
Should the limit be lowered to `EnableExplicitMT()` (i.e. when called with the default argument)? If 32bit and > 8 cores, cap it at 8 threads? Else we run the risk of exhausting the address space again soon enough with some other test, or on some user's analysis... What do you think? Or is this the user's responsibility?

File: tree/dataframe/test/dataframe_concurrency.cxx (hunk 1 above)
Comment (stwunsch) [reply tree/dataframe/test/dataframe_concurrency.cxx:22]: Fails also on 64bit with enough cores ;) What is the magic preproc var for detecting 32bit?

**Comment by stwunsch** [comment]
> Should the limit be lowered to `EnableExplicitMT()` (i.e. when called with the default argument)? If 32bit and > 8 cores, cap it at 8 threads? Else we run the risk of exhausting the address space again soon enough with some other test, or on some user's analysis... What do you think? Or is this the user's responsibility?

I have the strong opinion that it's the user's responsibility. Mainly because the limit of 8 threads is not necessarily a sane default, it depends on the program you execute concurrently.

File: tree/dataframe/test/dataframe_concurrency.cxx (hunk 1 above)
Comment (stwunsch) [reply tree/dataframe/test/dataframe_concurrency.cxx:22]: Found it, it's `R__B64`

Comment (Axel-Naumann) [reply tree/dataframe/test/dataframe_concurrency.cxx:22]: Can you `#include "ROOT/RConfig.hxx"` to make sure we see `R__B64` without relying on side-effects of other `#include`s?

**Review by pcanal — APPROVED** [review]

**Pull request merged**